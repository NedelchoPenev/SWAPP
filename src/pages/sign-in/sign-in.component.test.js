import React from 'react';
import { shallow } from 'enzyme';

import SignIn from './sign-in.component';
import FormInput from '../../components/form-input/form-input.component';

describe('SignIn component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SignIn />);
  });

  it('should render SignIn component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleChange on change event', () => {
    const event = {
      target: {
        name: 'email',
        value: 'test'
      }
    };

    wrapper.find(FormInput).at(0).prop('handleChange')(event)
    expect(wrapper.state('email')).toBe('test');
  });

  it('should call onSubmit when submit button clicked', () => {
    const preventDefault = { preventDefault: () => {} };
    wrapper.setProps({signIn: jest.fn()});
    wrapper.find('form').simulate('submit', preventDefault);
    expect(wrapper.state('submitted')).toBe(true);
  });

  it('should NOT show error message with right credentials', () => {
    wrapper.setProps({
      error: false
    })

    expect(wrapper.find('ErrorMessage')).toEqual({});
  });

  it('should show error message with wrong credentials', () => {
    wrapper.setProps({
      error: {
        graphQLErrors: [{
          message: 'test'
        }]
      }
    })

    expect(wrapper.find('ErrorMessage').at(0).text()).toBe('test');
  });
});