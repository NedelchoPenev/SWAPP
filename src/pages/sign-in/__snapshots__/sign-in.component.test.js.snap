// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component should render SignIn component 1`] = `
<styled.div>
  <LogoContainer
    asTitle={true}
  />
  <styled.div>
    <form
      onSubmit={[Function]}
    >
      <FormInput
        handleChange={[Function]}
        name="email"
        placeholder="Email"
        required={true}
        type="text"
        value=""
      />
      <FormInput
        handleChange={[Function]}
        name="password"
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      <CustomButton
        id="form-button"
        type="submit"
      >
         Login 
      </CustomButton>
    </form>
  </styled.div>
  <p>
    <span>
      Email: 
    </span>
    demo@st6.io
  </p>
  <p>
    <span>
      Password: 
    </span>
    demo1234
  </p>
</styled.div>
`;
